### DISCUSSION TITLE
Sign up with magic link and encounter AuthWeakPasswordError

### DISCUSSION THREAD

**Discussion by abbylow** [supabase/supabase#20024] [Questions] [answered]
Context: I'm using @supabase/ssr (version 0.0.10) and @supabase/supabase-js (version 2.39.1) with NextJs App Router. 

What I am trying to do: Let user sign in with magic link and if a user has not signed up yet, automatically sign up the user.

Issue: Magic link works well when an existing user signs in. However, if a new user signs in with magic link, it showed error that the password is not strong enough when there is no password submitted. 

Code Snippet: 
```    
const { data, error } = await supabase.auth.signInWithOtp({
      email,
      options: {
        // I want the user to be automatically signed up
        shouldCreateUser: true,
      },
    })
```

Error: 
```
signIn error  AuthWeakPasswordError: Password should contain at least one character of each: abcdefghijklmnopqrstuvwxyz, ABCDEFGHIJKLMNOPQRSTUVWXYZ, 0123456789, !@#$%^&*()_+-=[]{};\'\:"|<>?,./`~.
```

**Comment by GaryAustin1** [comment]
Probably related to the new password strength settings.
There is a bug filed:
https://github.com/supabase/gotrue/issues/1359

**Reply by abbylow** [reply]
Thanks for your prompt reply. 

In this case, should I use an older version of packages to bypass this issue first? Do you have any recommended versions that should be working?

**Reply by GaryAustin1** [reply, marked as answer]
Do you have enhanced password settings set?  
![image](https://github.com/supabase/supabase/assets/54564956/18e2b0fd-b2de-4c55-82e1-399f09490e6b)
If so turn those off.

You can't go backwards on this for Supabase hosted instances as it is done in the gotrue server and not the client.

**Reply by abbylow** [reply]
Oh you are right. Turn this password requirements off and the magic link sign-up works well now. 
I guess this password settings won't be an issue if I only allow user to sign up with magic link and Oauth. 

Thanks a lot Gary!
